ckanext/envvars/tests/test_gte_ckan210.py

```diff
@@ -0,0 +1,87 @@
+import os
+import pytest
+import ckan.plugins as p
+from ckantoolkit import config
+import ckantoolkit as toolkit
+
+
+if toolkit.check_ckan_version(max_version='2.9.99'):
+    pytest.skip("Skipping CKAN < 2.10", allow_module_level=True)
+
+
+from unittest import mock
+
+
+class TestCkanCoreEnvVarsConfig(object):
+
+    '''
+    Some values are are transformed into ini settings by core CKAN. These
+    tests makes sure they still work.
+    '''
+
+    def _setup_env_vars(self, envvar_list):
+        for env_var, value in envvar_list:
+            os.environ[env_var] = value
+        # plugin.load() will force the config to update
+        p.load()
+
+    def _teardown_env_vars(self, envvar_list):
+        for env_var, _ in envvar_list:
+            if os.environ.get(env_var, None):
+                del os.environ[env_var]
+        # plugin.load() will force the config to update
+        p.load()
+
+    @mock.patch('ckanext.datastore.plugin.DatastorePlugin.configure')
+    def test_core_ckan_envvar_values_in_config(self, datastore_configure):
+
+        if not toolkit.check_ckan_version('2.4.0'):
+            raise pytest.skip('CKAN version 2.4 or above needed')
+
+        core_ckan_env_var_list = [
+            ('CKAN_SQLALCHEMY_URL', 'postgresql://mynewsqlurl/'),
+            ('CKAN_DATASTORE_WRITE_URL', 'postgresql://mynewdbwriteurl/'),
+            ('CKAN_DATASTORE_READ_URL', 'postgresql://mynewdbreadurl/'),
+            ('CKAN_SITE_ID', 'my-site'),
+            ('CKAN_DB', 'postgresql://mydeprectatesqlurl/'),
+            ('CKAN_SMTP_SERVER', 'mail.example.com'),
+            ('CKAN_SMTP_STARTTLS', 'True'),
+            ('CKAN_SMTP_USER', 'my_user'),
+            ('CKAN_SMTP_PASSWORD', 'password'),
+            ('CKAN_SMTP_MAIL_FROM', 'server@example.com'),
+            ('CKAN__DATASETS_PER_PAGE', '14'),
+            ('CKAN__HIDE_ACTIVITY_FROM_USERS', 'user1 user2'),
+        ]
+
+        self._setup_env_vars(core_ckan_env_var_list)
+
+        assert config['sqlalchemy.url'] == 'postgresql://mynewsqlurl/'
+        assert config['ckan.datastore.write_url'] == 'postgresql://mynewdbwriteurl/'
+        assert config['ckan.datastore.read_url'] == 'postgresql://mynewdbreadurl/'
+        assert config['ckan.site_id'] == 'my-site'
+        assert config['smtp.server'] == 'mail.example.com'
+        assert config['smtp.starttls'] == 'True'
+        assert config['smtp.user'] == 'my_user'
+        assert config['smtp.password'] == 'password'
+        assert config['smtp.mail_from'] == 'server@example.com'
+        # I'll expect CKAN 2.10 to transform this to an int
```

Comment:
Rather than split the tests into separate modules by version let's keep them all in the same file but use `pytest.if` to separate these that have different outputs in different CKAN versions

Reply:
Done

Reply:
I still see the two test modules?

Reply:
It's rare, I rename the test files [here](https://github.com/okfn/ckanext-envvars/commit/f8f221e09601677f88c3a60f1eb7cd6fb1be68a2) but the PR continue showing old name files. Now I do it again [here](https://github.com/okfn/ckanext-envvars/commit/3c82b909269b8320d32ccc56bdb9df98a08b5a9f) but the PR continue showing bad file names. Checking